specification/loadtestservice/resource-manager/Microsoft.LoadTestService/stable/2022-12-01/loadtestservice.json

```diff
@@ -0,0 +1,963 @@
+        "Failed",
+        "Canceled",
+        "Deleted"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ResourceState",
+        "modelAsString": true
+      }
+    },
+    "QuotaResourceList": {
+      "description": "A list of Quota Bucket details. It contains an URL link to get the next set of results.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/QuotaResource"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "description": "List of Quota Bucket details by the loadtestservice resource provider",
+          "readOnly": true
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to get the next set of Quota Bucket details results (if there are any).",
+          "readOnly": true
+        }
+      }
+    },
+    "QuotaResource": {
+      "type": "object",
+      "description": "Quota Bucket details",
+      "properties": {
+        "properties": {
+          "description": "Quota resource properties",
+          "$ref": "#/definitions/QuotaResourceProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "QuotaResourceProperties": {
+      "description": "Quota resource properties.",
+      "type": "object",
+      "properties": {
+        "limit": {
+          "description": "Quota Limit of the quota bucket.",
+          "type": "integer",
+          "minimum": 0,
+          "format": "int32"
+        },
+        "usage": {
+          "description": "Current Usage of the quota bucket.",
+          "type": "integer",
+          "minimum": 0,
+          "format": "int32"
+        },
+        "provisioningState": {
+          "description": "Resource provisioning state.",
+          "$ref": "#/definitions/ResourceState",
+          "readOnly": true
+        }
+      }
+    },
+    "CheckQuotaAvailabilityResponse": {
+      "description": "Check Quota Availability Response.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "Check Quota Availability Response properties",
+          "$ref": "#/definitions/CheckQuotaAvailabilityResponseProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "CheckQuotaAvailabilityResponseProperties": {
+      "description": "Check Quota Availability Response properties.",
+      "type": "object",
+      "properties": {
+        "isAvailable": {
+          "description": "true if Quota is Available, else false.",
+          "type": "boolean"
+        },
+        "availabilityStatus": {
```

Comment:
If this property takes a fixed set of values it should be modeled as string enum, rather than unrestricted string. Also, it seems to overlap with the isAvailable property. What does this property contain?

Reply:
@mentat9 same as above.
Structure provided by QMS -
{
    "isAvailable": <<Boolean indicating whether the quota request be granted based on availability>>,
    "availabilityStatus": "message indicating additional details to add to quota support request"
}

https://eng.ms/docs/cloud-ai-platform/ahsi-organization/ahsi/cscp-teams/cscp-quotamanagement/azure-quota-services/policy-qms/resource-provider/ref-check-availability